sound/soc/sof/intel/hda.c

```diff
@@ -1039,13 +1039,72 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 #endif
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE)
+/* Check if all slaves defined on the link can be found */
+static bool link_slaves_found(struct snd_sof_dev *sdev,
+			      struct snd_soc_acpi_link_adr *link,
+			      struct sdw_intel_ctx *sdw)
+{
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct sdw_intel_slave_id *ids = sdw->ids;
+	int num_slaves = sdw->num_slaves;
+	unsigned int part_id, link_id, unique_id, mfg_id;
+	int j, k;
+
+	/*
+	 * Spec definition
+	 *   Register           Bit     Contents
+	 *   DevId_0 [7:4]      47:44   sdw_version
+	 *   DevId_0 [3:0]      43:40   unique_id
+	 *   DevId_1            39:32   mfg_id [15:8]
+	 *   DevId_2            31:24   mfg_id [7:0]
+	 *   DevId_3            23:16   part_id [15:8]
+	 *   DevId_4            15:08   part_id [7:0]
+	 *   DevId_5            07:00   class_id
```

Comment:
you need to add the bits in 48..63 that define the link mask for the _ADR.

Reply:
> you need to add the bits in 48..63 that define the link mask for the _ADR.

@plbossart I think it is link id, not link mask, right? For example, it is 0x3, but not 0x8 if the Slave is on link 3

static const u64 rt715_3_adr[] = {
	0x000**3**10025D071500
};

Reply:
yse, that's what I was referring to